python/lsst/pipe/base/task.py

```diff
@@ -111,7 +111,7 @@ def __init__(self, config=None, name=None, parentTask=None, log=None):
             if config is None:
                 config = getattr(parentTask.config, name)
             self._taskDict = parentTask._taskDict
-            self.log = Log.getLogger(parentTask.log.getName() + '.' + name)
+            loggerName = parentTask.log.getName() + '.' + name
```

Comment:
I don't think the logic or API is changed, and your commit is still correct.  Do I misunderstand @timj that separating the logger name operation from getting the logger
would make this clearer? 

If so, I'll remove this commit, and simply go back to remove the unnecessary `isinstance` check.